manifests/haproxy.pp

```diff
@@ -958,6 +968,23 @@
     }
   }
 
+  if $etcd {
+    haproxy::listen { 'etcd':
+      bind             => $etcd_bind_opts,
+      options          => {
+        'balance' => 'source',
+      },
+      collect_exported => false,
+    }
+    haproxy::balancermember { 'etcd':
+      listening_service => 'etcd',
+      ports             => '4001',
```

Comment:
No, 4001 is one of the 2 default ports client listens to so I just picked it, but perhaps 2379 is better since it's the official port. I'll change to 2379.